src/coreclr/jit/fgprofilesynthesis.cpp

```diff
@@ -31,8 +31,9 @@
 //
 void ProfileSynthesis::Run(ProfileSynthesisOption option)
 {
-    m_dfsTree = m_comp->fgComputeDfs();
-    m_loops   = FlowGraphNaturalLoops::Find(m_dfsTree);
+    m_dfsTree             = m_comp->fgComputeDfs();
+    m_loops               = FlowGraphNaturalLoops::Find(m_dfsTree);
+    m_improperLoopHeaders = m_loops->ImproperLoopHeaders();
```

Comment:
Nit: remove the field from `ProfileSynthesis` and just query it when necessary?

Reply:
Which field -- `m_dfsTree`?

Reply:
FYI `m_improperLoopHeaders` will get more uses in a follow-on PR.

Reply:
`ProfileSynthesis::m_improperLoopHeaders` uses can be replaced by `m_loops->ImproperLoopHeaders()`.